dietpi/dietpi-software

```diff
@@ -9878,6 +9922,42 @@ _EOF_
 
 		fi
 
+		software_id=186 # IPFS Node
+		if (( ${aSOFTWARE_INSTALL_STATE[$software_id]} == 1 )); then
+
+			Banner_Configuration
+
+			G_EXEC mkdir /mnt/dietpi_userdata/ipfs
+			Create_User -d /mnt/dietpi_userdata/ipfs ipfs
+			G_EXEC chown -R ipfs:ipfs /mnt/dietpi_userdata/ipfs
+
+			cat << '_EOF_' > /etc/systemd/system/ipfs.service
+[Unit]
+Description=IPFS (DietPi)
+Wants=network-online.target
+After=network-online.target dietpi-boot.service
+
+[Service]
+User=ipfs
+Environment=IPFS_PATH=/mnt/dietpi-userdata/ipfs
+ExecStart=/usr/local/bin/ipfs daemon
+
+[Install]
+WantedBy=multi-user.target
+_EOF_
+			G_EXEC systemctl daemon-reload
+			G_EXEC systemctl enable ipfs
+
+						cat << '_EOF_' > /mnt/dietpi_userdata/ipfs/config
+{
+	"Addresses.Gateway": "/ip4/127.0.0.1/tcp/8087"
+}
+_EOF_
```

Comment:
Let's no overwrite existing config files on reinstall, so prefix with `[[ -f '/mnt/dietpi_userdata/ipfs/config' ]]`. I guess the below init could be skipped as well when the config file exists already, so both can be wrapped into the file existence conditional?